kotlin/src/test/kotlin/com/linecorp/armeria/server/FlowAnnotatedServiceTest.kt

```diff
@@ -0,0 +1,278 @@
+                        }
+
+                        @Get("/custom-dispatcher")
+                        @ProducesText
+                        suspend fun dispatcherContext() = flow {
+                            assertThat(Thread.currentThread().name).contains("custom-thread")
+                            emit("OK")
+                        }
+
+                        @Get("/cancellation")
+                        @ProducesJsonSequences
+                        suspend fun cancellation() = flow {
+                            try {
+                                emit("OK")
+                                delay(2500L)
+                                emit("world")
+                            } catch (e: CancellationException) {
+                                cancelled.set(true)
+                            }
+                        }
+                    })
+                    decorator(
+                        Route.builder().path("/flow", "/custom-context").build(),
+                        CoroutineContextService.newDecorator { User(name = "Armeria", role = "Admin") }
+                    )
+                    decorator(
+                        Route.builder().path("/flow", "/custom-dispatcher").build(),
+                        CoroutineContextService.newDecorator {
+                            Executors
+                                .newSingleThreadExecutor { Thread(it, "custom-thread") }
+                                .asCoroutineDispatcher()
+                        }
+                    )
+                    requestTimeoutMillis(2000L)
+                }
+            }
+        }
+
+        lateinit var client: WebClient
+
+        @JvmStatic
+        @BeforeAll
+        @Suppress("unused")
+        fun init() {
+            client = WebClient.of(server.httpUri())
+        }
+    }
+}
+
+private val cancelled = AtomicBoolean()
+
+private data class Member(
+    val name: String,
+    val age: Int
+)
+
+private data class User(
+    val name: String,
+    val role: String
+) : AbstractCoroutineContextElement(User) {
+    companion object Key : CoroutineContext.Key<User>
+}
+
+internal infix fun HttpResponse.shouldProduce(expected: List<String>) {
+    val stream = expected.toMutableList()
+    val latch = CountDownLatch(1)
+    val cause = AtomicReference<Throwable?>()
+
+    this.split().body().subscribe(object : Subscriber<HttpData> {
+        lateinit var subscription: Subscription
+
+        override fun onSubscribe(s: Subscription) {
+            subscription = s
+            s.request(Long.MAX_VALUE)
+        }
+
+        override fun onNext(t: HttpData) {
+            assertThat(t.toStringUtf8()).isEqualTo(stream.removeAt(0))
+        }
+
+        override fun onError(t: Throwable) {
+            cause.set(t)
+            subscription.cancel()
+            latch.countDown()
+        }
+
+        override fun onComplete() {
+            latch.countDown()
+        }
+    })
+    latch.await()
```

Comment:
We prefer to use `await().untilTrue(AtomicBoolean)` for waiting for the completion of an event.
`latch.await()` will wait forever if something is wrong and the CI build will not be finished before the timeout of the CI job.

Reply:
Replaced it with `StepVerifier`.